R/misc.R

```diff
@@ -475,39 +475,93 @@ load_regional_functional_data <- function(...) {
 #' print(consolidated_weights)
 #' @import dplyr
 #' @export
-load_twas_weights <- function(weight_db_file, condition, 
+load_twas_weights <- function(weight_db_files, conditions = NULL,
                               variable_name_obj = "variant_names",
                               twas_weights_table = "twas_weights") {
   ## Internal function to load and validate data from RDS files
-  validate_data <- function(weight_db_file, condition, variable_name_obj) {
-    ## Check if the specified region and condition are available in all files
-    if (!(condition %in% names(weight_db_file[[1]]))) {
-        stop("The specified condition is not available in weight RDS file.")
+  load_and_validate_data <- function(weight_db_files, conditions, variable_name_obj) {
+    all_data <- lapply(weight_db_files, readRDS)
+    unique_regions <- unique(unlist(lapply(all_data,function(data) names(data))))
+    # Check if region from all RDS files are the same
+    if (length(unique_regions) != 1) {
+      stop("The RDS files do not refer to the same region.")
+    } else {
+    # Assuming all data refer to the same region, now combine data by conditions
+      combined_all_data <- do.call("c", lapply(all_data, function(data) data[[1]]))
     }
-    if (!is.null(variable_name_obj) && !(variable_name_obj %in% names(weight_db_file[[1]][[condition]]))) {
-        stop("The specified variable_name_obj is not available in weight RDS file.")
+    # Set default for 'conditions' if they are not specified
+    if (is.null(conditions)) {
+    conditions <- names(combined_all_data)
+    }                                           
+    ## Check if the specified condition and variable_name_obj are available in all files
+    if (!all(conditions %in% names(combined_all_data))) {
+        stop("The specified condition is not available in all RDS files.")
+      }
+    for (data in seq_along(combined_all_data))                                       
+    {
+        if (!is.null(variable_name_obj) && !(variable_name_obj %in% names(combined_all_data[[data]]))) {
```

Comment:
no need to chekc this if you use get_nested_elements. Please do that and remove these lines